pilz_robot_programming/test/integrationtests/tst_api_brake_test.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# Copyright (c) 2019 Pilz GmbH & Co. KG
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+import rospy
+
+from pilz_robot_programming.exceptions import RobotBrakeTestException
+from pilz_robot_programming.robot import Robot
+from brake_test_mock import BrakeTestMock
+from prbt_hardware_support.srv import BrakeTestResponse
+from prbt_hardware_support.msg import BrakeTestErrorCodes
+
+
+API_VERSION = "1"
+
+
+class TestAPIBrakeTest(unittest.TestCase):
+    """
+    Test brake test support in the python api.
+    """
+
+    def setUp(self):
+        self.robot = Robot(API_VERSION)
+
+    def tearDown(self):
+        if hasattr(self, 'robot'):
+            self.robot._release()
+            self.robot = None
+
+    def test_required(self):
+        """
+        We expect the method to return true, if the brake test is required.
+        """
+        mock = BrakeTestMock()
+        mock.start()
+        mock.advertise_brake_test_required_service()
+
+        mock.set_is_brake_test_required(True)
+        res = None
+        try:
+            res = self.robot.is_brake_test_required()
+        except Exception as e:
+            rospy.logerr(e)
+            raise e
```

Comment:
If everything is ok, no exception should be thrown, so why catch it here? Is it not easier and better (in this case at least) to simply ignore possible exceptions? If an exception raises, it will be automatically forwarded to the console.

Reply:
Removed except